packages/cli/src/server/api/memory/rooms.ts

```diff
@@ -0,0 +1,170 @@
+  params: {
+    agentId: string;
+    roomId?: string;
+  };
+}
+
+/**
+ * Room management functionality for agents
+ */
+export function createRoomManagementRouter(
+  agents: Map<UUID, IAgentRuntime>,
+  serverInstance: AgentServer
+): express.Router {
+  const router = express.Router();
+
+  // Create a new room for an agent
+  router.post('/:agentId/rooms', async (req, res) => {
+    const agentId = validateUuid(req.params.agentId);
+    if (!agentId) {
+      return sendError(res, 400, 'INVALID_ID', 'Invalid agent ID format');
+    }
+
+    const runtime = agents.get(agentId);
+    if (!runtime) {
+      return sendError(res, 404, 'NOT_FOUND', 'Agent not found');
+    }
+
+    try {
+      const { name, type = ChannelType.DM, source = 'client', worldId, metadata } = req.body;
+
+      if (!name) {
+        return sendError(res, 400, 'MISSING_PARAM', 'Room name is required');
+      }
+
+      const roomId = createUniqueUuid(runtime, `room-${Date.now()}`);
+      const serverId = req.body.serverId || `server-${Date.now()}`;
+
+      let resolvedWorldId = worldId;
+      if (!resolvedWorldId) {
+        const worldName = `World for ${name}`;
+        resolvedWorldId = createUniqueUuid(runtime, `world-${Date.now()}`);
+
+        await runtime.ensureWorldExists({
+          id: resolvedWorldId,
+          name: worldName,
+          agentId: runtime.agentId,
+          serverId: serverId,
+          metadata: metadata,
+        });
+      }
+
+      await runtime.ensureRoomExists({
+        id: roomId,
+        name: name,
+        source: source,
+        type: type,
+        channelId: roomId,
+        serverId: serverId,
+        worldId: resolvedWorldId,
+        metadata: metadata,
+      });
+
+      await runtime.addParticipant(runtime.agentId, roomId);
+      await runtime.ensureParticipantInRoom(runtime.agentId, roomId);
+      await runtime.setParticipantUserState(roomId, runtime.agentId, 'FOLLOWED');
+
+      sendSuccess(
+        res,
+        {
+          id: roomId,
+          name: name,
+          agentId: agentId,
+          createdAt: Date.now(),
+          source: source,
+          type: type,
+          worldId: resolvedWorldId,
+          serverId: serverId,
+          metadata: metadata,
+        },
+        201
+      );
+    } catch (error) {
+      logger.error(`[ROOM CREATE] Error creating room for agent ${agentId}:`, error);
+      sendError(res, 500, 'CREATE_ERROR', 'Failed to create room', error.message);
+    }
+  });
+
+  // Get all rooms where an agent is a participant
+  router.get('/:agentId/rooms', async (req, res) => {
+    const agentId = validateUuid(req.params.agentId);
+    if (!agentId) {
+      return sendError(res, 400, 'INVALID_ID', 'Invalid agent ID format');
+    }
+
+    const runtime = agents.get(agentId);
+    if (!runtime) {
+      return sendError(res, 404, 'NOT_FOUND', 'Agent not found');
+    }
+
+    try {
+      const worlds = await runtime.getAllWorlds();
```

Comment:
we could make a better SELECT WHERE agentId query instead of looping all worlds in JS